tests/test_util.py

```diff
@@ -78,7 +78,10 @@ def test_echo_result_echos_all_when_not_json(self, mock_click, mock_scanner):
         mock_scanner.exclude_signatures = []
         mock_scanner.issues = [1, 2, 3, 4]
         util.echo_result(options, mock_scanner, "", "")
-        mock_click.echo.assert_called_once_with("1\n2\n3\n4")
+        # This is the result of calling `bytes()` on integers.
+        mock_click.echo.assert_called_once_with(
+            b"\x00\n\x00\x00\n\x00\x00\x00\n\x00\x00\x00\x00"
+        )
```

Comment:
This doesn't thrill me because I can't make out any relationship between the input and output that convinces me we know what is supposed to happen or that the input even has an impact on the output.

What if you go back to the original code (original L79-L81) and change L79 to read

```python
mock_scanner.issues = [b"1", b"2", b"3", b"4"]
```
and then prefix the string in L81 with `b`?

Reply:
I can see the difficulty of interpreting this at a glance. The byte string we are asserting isn't very visually pleasant to parse. As to whether "we know what is supposed to happen", I would say that this test (still) is direct proof that we know. Specifically this is the results of `byte(1) + "\n" + byte(2) + "\n" + byte(3) + "\n" + byte(4)`.

I had initially tried to make this a little more pleasing to read, but this turned out to be the far easier option. That said, I can put some more effort into making this easier to visually parse.

Reply:
Would you find this easier to parse if I did something like this in the assert?
```pycon
>>> bytes(1) + b"\n" + bytes(2) + b"\n" + bytes(3) + b"\n" + bytes(4)
b'\x00\n\x00\x00\n\x00\x00\x00\n\x00\x00\x00\x00'
```

Reply:
Yes. Or alternatively add a comment explaining to the reader that we expect the UTF-8 encoding of integer N to be a string of N nul bytes. (Something I "know" only by reverse-engineering the observed output.)